### PR TITLE
 #30 add yardage calculator

### PR DESCRIPTION
**Pull request by li-grace**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)

Implemented Feature #(feature)

## How it was implemented

Please describe the technical details of how this change was implemented. Include any relevant code changes, libraries used, or other technical details.

## How it was tested

Please describe how this change was tested. Include any relevant testing frameworks or methodologies used, and any manual testing that was performed.

## Screenshots or video

Please include a screenshot or video of the change in action, if applicable.

### CODE REVIEW

File: api/index.py [hunk 1]
```diff
@@ -51,6 +54,12 @@ def post(self):
...
 
+        # GRACE-TODO: Process fabric width information in request
+        fabricWidth = request.form["width"]
+        fabricUnit = request.form["unit"]
+        resolution_manager.width = fabricWidth
+        resolution_manager.height = fabricWidth * 4
+
         # Process the uploaded file
         file = request.files["file"]
         fabricLength = request.form["length"]
```
Comment (Joshua-Pow) [inline api/index.py]: Remove these two lines since you handled it above

File: api/resolution.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        Calculate final pattern yardage requirement given array of polygons
+        Returns a tuple representing width, height in specified units
+        polygonArray: a list of Polygon types containing bbox_low_x/y and width/height
+        unit: cm or inches
+        """
+        pixel_h = self._get_bounding_box_length(self, polygonArray)
+        #real_width = self._get_real_world_yardage(self, pixel_w_h[0], unit) #only need the height for yardage
+        return pixel_h * self._get_scaling_factor(unit)
+
+    def _get_scaling_factor(self, unit="inch"):
```
Comment (Joshua-Pow) [inline api/resolution.py:55]: Why do these have names starting with _ ?

File: api/resolution.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        scale = get_scaling_factor(unit="inch")
+        actual_fabric_length = bounding_box_length * scale
+        """
+        if (unit[0]=="i"): #inch
+            return (output_height_inch / self.pdf_height)
+
+        return (self.output_height_cm / self.pdf_height)
+
+
+    def _get_bounding_box_length(self, polygonArray):
```
Comment (Joshua-Pow) [inline api/resolution.py]: Same here

File: api/resolution.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        returns max y coordinate in all polygon bounding boxes
+        polygonArray: a list of Polygon types containing bbox_low_x/y and width/height
+        """
+        current_length = 0
+
+        for polygon in polygonArray:
+            current_length = max(current_length,polygon.bbox_low_y + polygon.height)
+        return current_width
+
+    def _get_bounding_box_width_limit(self):
```
Comment (Joshua-Pow) [inline api/resolution.py:79]: and here

File: api/resolution.py (hunk 2 above)
Comment (li-grace) [reply api/resolution.py:55]: "hidden" / private function naming convention

File: api/index.py [hunk 5]
```diff
@@ -165,17 +174,22 @@ def get(self):
                 )
             )
 
-        # TODO: change x to be the width of the fabric
-        container_max_x = 2000
-        container_max_y = 2000
+        container_max_x = resolution_manager._get_bounding_box_width_limit()
```
Comment (Joshua-Pow) [inline api/index.py:177]: Im pretty sure for the _ notation with functions you're only supposed to use them within the class itself

**Review by Joshua-Pow — APPROVED** [review]

**Pull request merged**